metadata.go

```diff
@@ -51,3 +63,58 @@ func (m *AppImageMetadata) metadataForBuildpack(id string) BuildpackMetadata {
 	}
 	return BuildpackMetadata{}
 }
+
+func (m *CacheImageMetadata) metadataForBuildpack(id string) BuildpackMetadata {
+	for _, bpMd := range m.Buildpacks {
+		if bpMd.ID == id {
+			return bpMd
+		}
+	}
+	return BuildpackMetadata{}
+}
+
+func getAppMetadata(image image.Image, log *log.Logger) (AppImageMetadata, error) {
+	metadata := AppImageMetadata{}
+	contents, err := getMetadata(image, MetadataLabel, log)
+	if err != nil {
+		return metadata, err
+	}
+
+	if err := json.Unmarshal([]byte(contents), &metadata); err != nil {
+		log.Printf("WARNING: image '%s' has incompatible '%s' label\n", image.Name(), MetadataLabel)
+		return metadata, nil
+	}
+	return metadata, nil
+}
+
+func getCacheMetadata(image image.Image, log *log.Logger) (CacheImageMetadata, error) {
+	metadata := CacheImageMetadata{}
+	contents, err := getMetadata(image, CacheMetadataLabel, log)
+	if err != nil {
+		return metadata, err
+	}
+
+	if err := json.Unmarshal([]byte(contents), &metadata); err != nil {
+		log.Printf("WARNING: image '%s' has incompatible '%s' label\n", image.Name(), CacheMetadataLabel)
+		return metadata, nil
+	}
+	return metadata, nil
+}
+
+func getMetadata(image image.Image, metadataLabel string, log *log.Logger) (string, error) {
+	if found, err := image.Found(); err != nil {
+		return "", err
+	} else if !found {
+		log.Printf("WARNING: image '%s' not found or requires authentication to access\n", image.Name())
+		return "", nil
+	}
+	contents, err := image.Label(metadataLabel)
+	if err != nil {
+		return "", err
+	}
+	if contents == "" {
+		log.Printf("WARNING: image '%s' does not have '%s' label", image.Name(), metadataLabel)
+		return "metadata", nil
```

Comment:
Should this be `return "", nil`? Returning the string `"metadata"` seems confusing.